### PR TITLE
CHE-2230: Save preferences every 10 seconds

### PR DESCRIPTION
**Pull request by svor**
### What does this PR do?
Adds a repeating command that is saved user's preferences every 10 seconds. 
This is temporary solution. Should be properly fixed by https://github.com/eclipse/che/issues/5338

### What issues does this PR fix or reference?
https://github.com/codenvy/codenvy/issues/2230

### CODE REVIEW

**Review by dkulieshov — APPROVED** [review]

**Comment by benoitf** [comment]
I'm not sure it looks great even as a temporary solution. There is no hooks to do that on some workspace actions ? also 10seconds looks like very often and it's not configurable at all.

**Comment by skabashnyuk** [comment]
is this implemented over websocket or direct http call?

File: core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/impl/GsonJsonRpcComposer.java [hunk 1]
```diff
@@ -64,7 +64,7 @@
             JsonArray jsonArray = new JsonArray();
             for (int i = 0; i < paramsList.size(); i++) {
                 JsonElement jsonElement = (JsonElement)paramsList.get(i);
-                jsonArray.set(i, jsonElement);
+                jsonArray.add(jsonElement);
```
Comment (skabashnyuk) [inline core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/impl/GsonJsonRpcComposer.java:67]: what is it? Fix of  some issue in RPC transport?

Comment (svor) [reply core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/impl/GsonJsonRpcComposer.java:67]: Yes, it is small fix for the JSON RPC composer

**Comment by svor** [comment]
@benoitf I have no idea which ws action I can use here (when browser tab with ide closes or reloads). If use windowClosing action for some user who has a big text of preferenses, it can't be saved because the browser canceles this request. 
So, @slemeur I don't think that this issue is regression and we need to fix this https://github.com/eclipse/che/issues/5338.
    
@skabashnyuk It is http call. I've tried to use JSON RPC but have found a bug (https://github.com/eclipse/che/issues/5369).

**Comment by dkulieshov** [comment]
Hi,

Hi, as of what I see this is not a regression as this is reproduced on a specific environment and I'm pretty sure that it was not tested on such environments earlier. This seems to be a downside of how we manage user preferences at the moment:
 - We have no possibility to incrementally save user preference data, so we send it all (dozens of KB)
 - We cannot confidently rely on hooks or any other event based mechanism (like close tab action, refresh, etc.) here

So we simply can't update  large user preferences properly and in time when we refresh/close tabs. Also taking into account time limit for this issue it is obvious that [there's no easy way out](https://www.youtube.com/watch?v=rOXaPE6gklI) and I see that the option that is chosen as the most appropriate here is to send it by schedule.

> I'm not sure it looks great even as a temporary solution. 

I can't agree more here. The solution that is presented is quite doubtful however here we have not many options. We either accept this as best afford or leave it as is and wait until user preferences implementation is updated/redesigned. I guess https://github.com/eclipse/che/issues/5338 is not enough for that, will create appropriate issue.

As this is not a regression and not a blocker I would humbly suggest to postpone it until proper fix is ready.

**Comment by riuvshin** [comment]
@svor so it seems we have to close this PR

**Comment by slemeur** [comment]
It is fine for me.

Please @svor and @dkuleshov, cross link the issue with the refactoring needed for preferences.

**Pull request closed without merging**